spec/unit/component_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'dry/system/component'
+
+RSpec.describe Dry::System::Component do
+  subject(:component) { Dry::System::Component.new(name, options) }
+
+  let(:options) do
+    { namespace_separator: namespace_separator, path_separator: path_separator }
+  end
+
+  let(:namespace_separator) { '.' }
+  let(:path_separator) { '/' }
+
+  context 'with default separators' do
+    let(:name) { :foo }
+
+    describe '#identifier' do
+      it 'returns qualified identifier' do
+        expect(component.identifier).to eql('foo')
+      end
+    end
+
+    describe '#namespaces' do
+      it 'returns namespace array' do
+        expect(component.namespaces).to eql(%i[foo])
+      end
```

Comment:
Wouldn't we want namespaces to be `[]` for a component with a single name not nested under anything?